### PR TITLE
chore: remove outdated editor grammar files

### LINKED ISSUE #398: Remove outdated files from top-level `grammar` folder
Essentially everything, except `configSchema.json`. [tact-sublime](https://github.com/tact-lang/tact-sublime) has now the most up-to-date grammar to use in editors

### PR DESCRIPTION
**Pull request by anton-trunov**
and rename the top-level `grammar/` folder into json-schemas: we might want to add more schemas for the JSON files Tact produces or uses.

Closes #398

- [x] I have updated CHANGELOG.md
- [x] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Comment by anton-trunov** [comment]
Related tact-docs issue: https://github.com/tact-lang/tact-docs/issues/256

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Trailing semicolons in struct and message declarations are optional now: PR [#395](https://github.com/tact-lang/tact/pull/395)
 - Tests are refactored and renamed to convey the sense of what is being tested and to reduce the amount of merge conflicts during development: PR [#402](https://github.com/tact-lang/tact/pull/402)
+- The outdated TextMate-style grammar files for text editors have been removed (the most recent grammar files can be found in the [tact-sublime]() repo): PR [#404](https://github.com/tact-lang/tact/pull/404)
```
Comment (novusnota) [inline CHANGELOG.md]:
```suggestion
- The outdated TextMate-style grammar files for text editors have been removed (the most recent grammar files can be found in the [tact-sublime](https://github.com/tact-lang/tact-sublime) repo): PR [#404](https://github.com/tact-lang/tact/pull/404)
```

**Review by novusnota — COMMENTED** [review]
All seems good. However, we can probably name the folder `schemas` (not `json-schemas`) for simplicity and a bit of future-proofing :)

**Comment by anton-trunov** [comment]
> All seems good. However, we can probably name the folder schemas (not json-schemas) for simplicity and a bit of future-proofing :)

done

**Pull request merged**